cmd/snap/main.go

```diff
@@ -423,6 +426,27 @@ func resolveApp(snapApp string) (string, error) {
 	return snapApp, nil
 }
 
+// exitCodeFromError takes an error and returns specific exit codes
+// for some errors. Otherwise the generic exit code 1 is returned.
+func exitCodeFromError(err error) int {
+	var mksquashfsError squashfs.MksquashfsError
+	var cmdlineFlagsError *flags.Error
+	var unknownCmdError unknownCommandError
+
+	switch {
+	case err == nil:
+		return 0
+	case xerrors.As(err, &cmdlineFlagsError) || xerrors.As(err, &unknownCmdError):
+		return 5
```

Comment:
There's a little known `sysexits.h` which defines the following constants:
```
#define EX_USAGE    64  /* command line usage error */
#define EX_DATAERR  65  /* data format error */
#define EX_NOINPUT  66  /* cannot open input */
#define EX_NOUSER   67  /* addressee unknown */
#define EX_NOHOST   68  /* host name unknown */
#define EX_UNAVAILABLE  69  /* service unavailable */
#define EX_SOFTWARE 70  /* internal software error */
#define EX_OSERR    71  /* system error (e.g., can't fork) */
#define EX_OSFILE   72  /* critical OS file missing */
#define EX_CANTCREAT    73  /* can't create (user) output file */
#define EX_IOERR    74  /* input/output error */
#define EX_TEMPFAIL 75  /* temp failure; user is invited to retry */
#define EX_PROTOCOL 76  /* remote error in protocol */
#define EX_NOPERM   77  /* permission denied */
#define EX_CONFIG   78  /* configuration error */
```

Perhaps this is something we could use (or not, just mentioning this to raise awareness).